cluster.go

```diff
@@ -70,10 +70,10 @@ func (vl VMList) Less(i, j int) bool { return vl[i].Name < vl[j].Name }
 
 func namesFromVMName(name string) (string, string, error) {
 	parts := strings.Split(name, "-")
-	if len(parts) != 3 {
+	if len(parts) < 3 {
 		return "", "", fmt.Errorf("expected VM name in the form %s, got %s", vmNameFormat, name)
 	}
-	return parts[0], parts[0] + "-" + parts[1], nil
+	return parts[0], strings.Join(parts[0:len(parts)-1], "-"), nil
```

Comment:
FYI, the `0:` isn't necessary. This could be `strings.Join(parts[:len(parts)-1], "-")`.

Reply:
Done